src/data/markdown/translated-guides/en/03 Results output/100 End-of-test/150-custom-summary.md

```diff
@@ -1,35 +1,76 @@
 ---
 title: Custom summary
-excerpt: With the handle summary, you can customize every part of your report, 
+excerpt: With the handlesummary() function, you can customize every part of your report. Change the content, redirect output, and more.
 ---
 
-Use the `handleSummary()` function to completely customize the end-of-test summary report.
-
-k6 calls`handleSummary()` at the end of the test run, even after [`teardown()`](/using-k6/test-life-cycle).
-If `handleSummary()` is exported, k6 does _not_ print the default summary.
-
-Besides customizing the CLI summary, you can also transform the summary data into machine- or human-readable formats.
-This lets you create JS-helper functions that generate JSON, CSV, XML (JUnit/xUnit/etc.), HTML, etc. files from the summary data.
+With the `handleSummary()` callback, you can completely customize your end-of-test summary.
```

Comment:
```suggestion
With the `handleSummary()` callback, you can completely customize your end-of-test summary.
```

Is there any reason to specify that it's a "callback" function? Or is this exposing too much internals? What function does it call back to?

If it's necessary to mention that it is a callback, maybe we could add an explanatory sentence at the end of the "About handleSummary()) section.

Reply:
I am for Less is More. So, in this context, it could be unnecessary. 

> With `handleSummary()`, you can 


[Read more about callbacks](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Introducing#callbacks)

Reply:
I would argue it shouldn't be called callback function.

From the [MDN article](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function) the definition is 
> a function passed into another function as an argument

We are missing the whole "passed into another function" so IMO we can drop it.

If we don't I would argue we should call `setup`, `teardown` and `default` callback functions as well ;)

Reply:
I'd separate this convo in another thread. Here my thoughts:


1 - I think `default` would be different --> the default scenario function?

2 - But I agree `setup`, `teardown`, and `handleSummary` follow the same pattern.



3.1 - I have seen using `callback` for this case. But it is confusing.
3.2 - Another common term in JS land is [hook](https://mochajs.org/#hooks)
3.3 - To keep it simple - I am leaning toward using the term `global functions`?

Reply:
Yes, this is beyond the scope of this PR, but I have two questions:

These functions are call backs to what other function? So it seems like these functions  run within a "super-function" that k6 automatically sets up when the script runs?

Is this detailed information about runtime useful to any users? If it is, we could probably add it to the Test life cycle document.

Reply:
> these functions run within a "super-function" that k6 automatically sets up when the script runs?

This won't be how I will explain it. K6 just happens to be calling this functions in a particular pattern that can be explained in the Test life cycle.

> I am leaning toward using the term global functions?

I am for "exported function"  as it (and the rest) need to be *exported* and a *function* so it seems like the best choice to me.

Reply:
Sorry, just for my understanding, isn't there some kind of "superfunction" under the hood? I mean, _to what_ are these functions exported?

I know this is way too technical, and I beg forgiveness for my poor vocabulary, but it has confused me for a while. I think it's related to how I got so confused by "where" the data in `handlesummary(data)` came from.

Reply:
I think the term "exported function" lacks some more context as any k6/JS module can export a function. 

Moving the convo to https://github.com/grafana/k6-docs/issues/845

Reply:
> Sorry, just for my understanding, isn't there some kind of "superfunction" under the hood? I mean, to what are these functions exported?

They are exported as [defined](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export) by the ecmascript specification. Which means that other JS modules can use them. 

k6 is not JS code it just uses the same mechanic to call them. 

I guess you can call it a superfunction (there is no such term, and it is not a [higher-order function](https://en.wikipedia.org/wiki/Higher-order_function)) it is just that k6 calls those functions with given parameters in a given ... pattern. They are [called](https://github.com/grafana/k6/blob/8a74171c8b43f48fb53078e0bcfa7851828124bc/js/runner.go#L321) from things that *are* functions within the go language :shrug:. 

But I am not really certain how this is relevant to anyone :shrug: 


>  got so confused by "where" the data in handlesummary(data) came from.

It comes from k6 providing it. 

In practice *almost* everything in computers comes from another program. When you run your program your OS provides it with a bunch of stuff in order to do it. In the same way we provide the info to the script when we execute it. It is just that OSes have some "standards" on how they will execute a program and we have some "standards" on how we will execute the script. And for the most parts most of the users do not care about *most* of those. 

In the cases of the functions we are talking about, you need to know that the first argument will be the data returned from setup or in the case of handleSummary the "summary data". Both of those are just things that when we designed the system decided will be useful. 

To be honest if we make this today - there is a good chance that some of those will just be things `k6/execution` will be able to return to you instead. As documenting that seems a lot easier and is definitely less confusing. Or at least we would have a discussion about it :shrug: 

Hope this helps you!